lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1916,284 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
+) -> None:
+    keyvault_client = get_key_vault_management_client(platform)
+
+    keyvault_poller = keyvault_client.vaults.delete(resource_group_name, vault_name)
+    log.debug(f"{keyvault_poller} - Key Vault {vault_name} deleted successfully.")
+
+
+def get_certificate_client(
+    vault_url: str, platform: "AzurePlatform"
+) -> CertificateClient:
+    return CertificateClient(vault_url, platform.credential)
+
+
+def get_secret_client(vault_url: str, platform: "AzurePlatform") -> SecretClient:
+    return SecretClient(vault_url, platform.credential)
+
+
+def create_certificate(
+    platform: "AzurePlatform",
+    vault_url: str,
+    log: Logger,
+    cert_name: str,
+    retries: int = 5,
+    delay: int = 2,
+) -> str:
+    certificate_client = get_certificate_client(vault_url, platform)
+    secret_client = get_secret_client(vault_url, platform)
+
+    cert_policy = CertificatePolicy.get_default()
+
+    for _attempt in range(retries):
+        # Create certificate
+        create_certificate_result = certificate_client.begin_create_certificate(
+            cert_name, policy=cert_policy
+        )
+        log.info(
+            f"Certificate '{cert_name}' has been created. "
+            f"Result: {create_certificate_result}"
+        )
+        certificate_client.update_certificate_properties(
+            certificate_name=cert_name, enabled=True
+        )
+        break
+
+    secret_id: Optional[str] = secret_client.get_secret(name=cert_name).id
+    if secret_id:
+        secret_url_without_version = secret_id.rsplit("/", 1)[0]
+        return secret_url_without_version
+    else:
+        raise LisaException(f"Failed to retrieve secret ID:'{cert_name}'.")
+
+
+def check_certificate_existence(
+    vault_url: str, cert_name: str, log: Logger, platform: "AzurePlatform"
+) -> bool:
+    certificate_client = CertificateClient(
+        vault_url=vault_url, credential=platform.credential
+    )
+
+    try:
+        certificate = certificate_client.get_certificate(cert_name)
+        log.info(f"Cert found '{certificate.name}'")
+        return True
+    except Exception as e:
+        if "not found" in str(e).lower():
+            log.info(f"Certificate '{cert_name}' does not exist.")
+            return False
+        else:
+            # Directly raise an exception without logging an error
+            raise LisaException(
+                f"Unexpected error checking certificate '{cert_name}': {e}"
+            )
+
+
+def delete_certificate(
+    platform: "AzurePlatform",
+    vault_url: str,
+    cert_name: str,
+    log: Logger,
+) -> bool:
+    certificate_client = get_certificate_client(vault_url, platform)
+
+    try:
+        certificate_client.begin_delete_certificate(cert_name)
+        log.info(f"Certificate {cert_name} deleted successfully.")
+        return True
+    except Exception as e:
+        log.error(f"Error deleting certificate {cert_name}: {e}")
```

Comment:
As previous comments, don't use log.error. If it's an error, raise an exception. If it's not, print a debug level log. Please update similar places.